src/Catalog/CatalogWriterBase.cs

```diff
@@ -18,8 +18,13 @@ public abstract class CatalogWriterBase : IDisposable
     {
         protected List<CatalogItem> _batch;
         protected bool _open;
+        protected string _customDomain;
 
-        public CatalogWriterBase(IStorage storage, ICatalogGraphPersistence graphPersistence = null, CatalogContext context = null)
+        public CatalogWriterBase(
```

Comment:
The feed2catalog will not be impacted by these changes as long as AppendOnlyCatalogWriter (used by feed2catalog) will not subscribe to use a customDomain.
By default customDomain is null and the ReplaceHostForUri is noop for this case. I will go some feed2catalog more testing.